src/clone-element.js

```diff
@@ -7,9 +7,13 @@ import { createVNode } from './create-element';
  * @param {import('./internal').VNode} vnode The virtual DOM element to clone
  * @param {object} props Attributes/props to add when cloning
  * @param {Array<import('./index').ComponentChildren>} rest Any additional arguments will be used as replacement children.
- * @returns {import('./internal').VNode}
+ * @returns {import('./internal').VNode|import('./index').ComponentChildren}
  */
 export function cloneElement(vnode, props) {
+	if (!vnode.type && !vnode._children) {
```

Comment:
Is this meant for supporting `text` vnodes? What's making this fail in the current scenario? Since this code isn't actually cloning it's still returning an equal reference making this node never render

Reply:
It would be meant for handling cases where `vnode` is of type `object | string | number | boolean | null | undefined`. 

Not complete sure... prior to this, I think passing children of those types would create an incomplete `vnode`.

```jsx
import { render, cloneElement } from 'preact';

const Foo = (props) => <div>{props.children}</div>;
const myFoo = <Foo>test</Foo>;
const clonedFoo = cloneElement(myFoo.props.children);
console.log(clonedFoo);
// type: undefined, props: {…}, key: undefined, ref: undefined, __k: null, …}
```

and you're right we would want to return the reference 😅

Reply:
This feels correct `// type: undefined, props: {…}, key: undefined, ref: undefined, __k: null, …}` because that's a textual vnode

Reply:
Thanks for the explanation, if I understand correctly (please correct me haha), then should changes should be done inside `createVNode`? or should we change how we reassign `props` within `cloneElement`?

Reply:
The latter sounds most correct to me, when we see the `type = null` we can just do `props = prevProps`